### ISSUE TITLE
Defining a space with multiple inputs

### ISSUE THREAD

**Issue by guillemhub** [openai/gym#995]
I am using keras-rl and gym using DQN for an autonomous drone.

Right now I have a state that is represented as a  30x100 image but I want to add more features to the state (Velocity, distance to destination, etc). One idea that I had was filling this new features as a new matrix 10x100 and then concatenate this obtaining a 40x100 matrix. But I think this won't be optimal because I am using the Atari DQN network, which performs convolutions to the image, and I probably will loose this new features or will be ignored for the neural network.

```
INPUT_SHAPE = (30, 100)
WINDOW_LENGTH = 1
input_shape = (WINDOW_LENGTH,) + INPUT_SHAPE

model = Sequential()
model.add(Conv2D(32, (4, 4), strides=(4, 4) ,activation='relu', input_shape=input_shape, data_format = "channels_first"))
model.add(Conv2D(64, (3, 3), strides=(2, 2),  activation='relu'))
model.add(Conv2D(64, (1, 1), strides=(1, 1),  activation='relu'))
model.add(Flatten())
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(nb_actions))
model.add(Activation('linear'))
print(model.summary())
```
I think the most optimal option will be defining the observation space and space in Gym in a different manner, and then merge or do something to avoid performing a Convolution to this new features, but I really do not know how to do it. Is there some info regarding this? My code defining the state in Gym is this.
```
self.observation_space = spaces.Box(low=0, high=255, shape=(30, 100))
self.state = np.zeros((30, 100), dtype=np.uint8)  
```
My full project can be found here:
https://github.com/guillem74/DRLDBackEnd

Thanks in advance.

**Comment by AloshkaD** [comment]
@guillem74 I'm familiar with both your work and @Kjell-K 's . However I'm not using keras-rl because I could not pass two images and sensor data and had many errors that took time to figure out until I gave up. I'm also having you both cited in my work to give credit.  
Anyway, I don't see where you define  an observation_space, is this an updated code that you have there? 
I might endup concatenating two images and padding the signal data to have an image channel shape and concatenate it as well and pass them all as one 7 channel image to the state and process them at the cnn door step.

**Comment by anthonyivol** [comment]
@AloshkaD have look at this gist : https://gist.github.com/bklebel/e3bd43ce228a53d27de119c639ac61ee
You might have to made few changes in keras to make it works (as describe in comments), but that allow to use dict/multi for observation space

**Comment by christopherhesse** [comment]
You can use a dictionary space for this, or else represent your other observations as colored boxes next to the original image.
